frontend/public/module/k8s/cluster-settings.ts

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable no-unused-vars, no-undef */
+import * as _ from 'lodash-es';
+
+import { ClusterVersionKind, ClusterUpdate, ClusterVersionConditionType, K8sResourceConditionStatus, ClusterVersionCondition } from '.';
+
+export enum ClusterUpdateStatus {
+  UpToDate = 'Up to Date',
+  UpdatesAvailable = 'Updates Available',
+  Updating = 'Updating',
+  Failing = 'Failing',
+  ErrorRetrieving = 'Error Retrieving',
+}
+
+export const getAvailableClusterUpdates = (cv: ClusterVersionKind): ClusterUpdate[] => {
+  return _.get(cv, 'status.availableUpdates', []);
+};
+
+export const getAvailableClusterChannels = () => ({'nightly-4.0': 'nightly-4.0', 'pre-release-4.0': 'pre-release-4.0', 'stable-4.0': 'stable-4.0'});
```

Comment:
These are supposed to be 1:1?

Reply:
Oh, yeah. This was actually part of @rhamilto's PR that added the update channel modal. I just moved it to this file. But it's like that because our dropdown component requires an object. If you pass an array, you get runtime errors in the logs.